controller/controller.go

```diff
@@ -3,41 +3,343 @@ package controller
 import (
 	"bytes"
 	"encoding/json"
+	"log"
 	"net/http"
+	"reflect"
 
-	service "github.com/raymundo/academy-go-q42021/service/csv"
+	"github.com/rayl17/academy-go-q42021/usecase"
 
 	"github.com/gorilla/mux"
 )
 
-func GetPokemonByName(w http.ResponseWriter, r *http.Request) {
+type ControllerInterface interface {
+	GetPokemonByName(w http.ResponseWriter, r *http.Request)
+	GetPokemonById(w http.ResponseWriter, r *http.Request)
+	PostPokemon(w http.ResponseWriter, r *http.Request)
+}
+
+type PokemonController struct {
+	usecase usecase.Usecaseinterface
+}
+
+func NewController(uc usecase.Usecaseinterface) *PokemonController {
+
+	return &PokemonController{usecase: uc}
+
+}
+
+func (c *PokemonController) GetPokemonByName(w http.ResponseWriter, r *http.Request) {
 	vars := mux.Vars(r)
 	name := vars["name"]
 
-	for _, pokemon := range service.GetPokemons() {
-		if pokemon.Name == name {
-			json.NewEncoder(w).Encode(pokemon)
-			return
+	if reflect.TypeOf(name).Kind() != reflect.String {
+		encoderErr := json.NewEncoder(w).Encode(map[string]string{"error": "Name should be an string"})
+		if encoderErr != nil {
+			log.Println(encoderErr)
 		}
 	}
 
-	var buffer bytes.Buffer
-	buffer.WriteString(`{message:"Pokemon not found"}`)
-	json.NewEncoder(w).Encode(map[string]string{"message": "Pokemon not found"})
+	pokemon, err := c.usecase.GetPokemonByName(name)
+
+	if err != nil {
+		var buffer bytes.Buffer
+		buffer.WriteString(`{message:"Pokemon not found"}`)
+
+		encoderErr := json.NewEncoder(w).Encode(map[string]string{"message": "Pokemon not found"})
+
+		if encoderErr != nil {
+			log.Println(encoderErr)
+		}
+
+	} else {
+		encodeErr := json.NewEncoder(w).Encode(*pokemon)
+
+		if encodeErr != nil {
+			log.Println(encodeErr)
+		}
+	}
 
 }
 
-func GetPokemonById(w http.ResponseWriter, r *http.Request) {
+func (c *PokemonController) GetPokemonById(w http.ResponseWriter, r *http.Request) {
```

Comment:
Acronyms should be all caps.

Also, remember to add documentation to your code by using comments; refer to https://go.dev/blog/godoc for further information. And please, apply these documentation comments to all exported functions and variables.
```suggestion
func (c *PokemonController) GetPokemonByID(w http.ResponseWriter, r *http.Request) {
```